### PR TITLE
Use a temporary table to verify if deletes are supported

### PR DESCRIPTION
**Pull request by hashhar**
Not all connector tests run within containers so any operation that
deletes data from the TPCH tables can change state of the testing
infrastructure leading to hard to diagnose failures.

So we create temporary tables to verify if deletes are supported or not.

### CODE REVIEW

**Comment by hashhar** [comment]
Follow-up to https://github.com/trinodb/trino/pull/8482.

**Review by wendigo — APPROVED** [review]

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java [hunk 1]
```diff
@@ -169,14 +170,16 @@ public void testInsert()
...
-        if (!hasBehavior(SUPPORTS_DELETE)) {
-            assertQueryFails("DELETE FROM region", "This connector does not support deletes");
-            return;
-        }
-
-        if (!hasBehavior(SUPPORTS_ROW_LEVEL_DELETE)) {
-            assertQueryFails("DELETE FROM region WHERE regionkey = 2", ".*[Dd]elet(e|ing).*(not |un)supported.*");
-            return;
+        skipTestUnless(hasBehavior(SUPPORTS_CREATE_TABLE));
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_delete", "(col bigint)", ImmutableList.of("1", "2"))) {
```
Comment (losipiuk) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:174]: can we just use `String tableName = "test_delete_" + randomTableSuffix();` for both "supports" and "unsupports" branches?

Comment (hashhar) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java:174]: Doesn't `TestTable` automatically add the suffix?

**Review by kokosing — APPROVED** [review]

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java (hunk 1 above)
Comment (hashhar) [reply testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorSmokeTest.java]: Oh, understood. AC.

**Review by lukasz-walkiewicz — APPROVED** [review]

**Comment by hashhar** [comment]
CI hit https://github.com/trinodb/trino/issues/4568.

**Pull request merged**